src/chemistry_functionality.jl

```diff
@@ -101,31 +101,29 @@ function make_compound(expr)
     species_expr = expr.args[2]
     species_name, ivs, _, _ = find_varinfo_in_declaration(expr.args[2])
 
-    # If no ivs were given, inserts `(..)` (e.g. turning `CO` to `CO(..)`).
-    isempty(ivs) && (species_expr = insert_independent_variable(species_expr, :(..)))
-
-    # Expression which when evaluated gives a vector with all the ivs of the components.
-    ivs_get_expr = :(unique(reduce(
-        vcat, [sorted_arguments(ModelingToolkit.unwrap(comp))
-               for comp in $components])))
+    # If no ivs were given, inserts  and expression which evaluates to the union of the ivs
+    # for the species the compound depends on.
+    ivs_get_expr = :(unique(reduce( vcat, [sorted_arguments(ModelingToolkit.unwrap(comp))
+        for comp in $components])))
```

Comment:
```suggestion
    ivs_get_expr = :(unique(reduce( vcat, (sorted_arguments(ModelingToolkit.unwrap(comp))
        for comp in $components))))
```
Does a generator work here instead?

Reply:
Probably would, will try and see